hydra/hydra.py

```diff
@@ -49,355 +53,488 @@ def __get_api(self, endpoint, parameters=None, headers=None):
+            auth=authentication,
+            json=payload,
+            verify=self.ca_certificate_path,
+        )
         if r.status_code != 200:
-            raise Exception('''Putting information to '{}' failed.\n
-                    Error Code: {}'''.format(endpoint, r.status_code))
-
+            raise Exception(
+                """Putting information to '{}' failed.\n
+                    Error Code: {}""".format(
+                    endpoint, r.status_code
+                )
+            )
 
     def __post_api(self, endpoint, parameters=None, payload=None):
-        authentication=(self.username, self.password)
-
-        r = requests.post("{}/{}".format(self.base_api_uri, endpoint),
-                params=parameters, auth=authentication, json=payload,
-                verify=self.ca_certificate_path)
+        authentication = (self.username, self.password)
+
+        r = requests.post(
+            "{}/{}".format(self.base_api_uri, endpoint),
+            params=parameters,
+            auth=authentication,
+            json=payload,
+            verify=self.ca_certificate_path,
+        )
         if r.status_code != 200:
             print(r.text)
-            raise Exception('''Posting information to '{}' failed.\n
-                    Error Code: {}'''.format(endpoint, r.status_code))
+            raise Exception(
+                """Posting information to '{}' failed.\n
+                    Error Code: {}""".format(
+                    endpoint, r.status_code
+                )
+            )
 
         return r.json()
 
-
     def __del_api(self, endpoint, parameters=None, payload=None):
-        authentication=(self.username, self.password)
-
-        r = requests.delete("{}/{}".format(self.base_api_uri, endpoint),
-                params=parameters, auth=authentication, json=payload,
-                verify=self.ca_certificate_path)
+        authentication = (self.username, self.password)
+
+        r = requests.delete(
+            "{}/{}".format(self.base_api_uri, endpoint),
+            params=parameters,
+            auth=authentication,
+            json=payload,
+            verify=self.ca_certificate_path,
+        )
         if r.status_code != 200:
             print(r.text)
-            raise Exception('''Deleting information from '{}' failed.\n
-                    Error Code: {}'''.format(endpoint, r.status_code))
+            raise Exception(
+                """Deleting information from '{}' failed.\n
+                    Error Code: {}""".format(
+                    endpoint, r.status_code
+                )
+            )
         try:
             response = r.json()
         except:
             response = r.text
-        if response == '': response = 'Deleting information from "{}" succeeded.'.format(endpoint)
+        if response == "":
+            response = 'Deleting information from "{}" succeeded.'.format(endpoint)
         return response
 
+    def deprecated(func):
```

Comment:
I'm not sure we want this as part of the class. 
Depending on how this grows having this as part of the class, may not make sense. 

@davemulford what are your thoughts here?

Reply:
That is fine with me. I am curious of Dave's thoughts, but it can be moved outside of the class easily.

Reply:
Per irc conversation with @davemulford, I moved from custom code for this over to [this prebuilt library](https://pypi.org/project/deprecation/).
This removes the need for the conversation wrt location of the function.